packages/pass-style/src/types.d.ts

```diff
@@ -80,20 +80,54 @@ export type PassByRef =
  * using 'slots').
  */
 export type Passable<
-  PC extends PassableCap = PassableCap,
+  R extends RemotableObject = RemotableObject,
   E extends Error = Error,
-> = void | Primitive | Container<PC, E> | PC | E;
-
-export type Container<PC extends PassableCap, E extends Error> =
-  | CopyArrayI<PC, E>
-  | CopyRecordI<PC, E>
-  | CopyTaggedI<PC, E>;
-interface CopyArrayI<PC extends PassableCap, E extends Error>
-  extends CopyArray<Passable<PC, E>> {}
-interface CopyRecordI<PC extends PassableCap, E extends Error>
-  extends CopyRecord<Passable<PC, E>> {}
-interface CopyTaggedI<PC extends PassableCap, E extends Error>
-  extends CopyTagged<string, Passable<PC, E>> {}
+  AllowPromise extends boolean = any,
+  AllowTopLevelPromise extends boolean = AllowPromise,
```

Comment:
Where do you make use of this fourth parameter?

Reply:
Internally to avoid including `PromiseLike` in the union of acceptable types inside the `PromiseLike` when recursing the type.